compiler/optimizer/VPConstraint.hpp

```diff
@@ -958,6 +958,8 @@ class VPKnownObject : public TR::VPFixedClass
    virtual TR::VPConstraint *intersect1(TR::VPConstraint *other, OMR::ValuePropagation *vp);
 
    virtual TR_YesNoMaybe isJavaLangClassObject();
+   virtual TR_YesNoMaybe isArrayWithConstantElements();
+   virtual void setIsArrayWithConstantElements();
```

Comment:
Setters on constraints are generally a bad idea - we need to create a new constraint to mutate it - I'd prefer to handle the extra parameter in the constructor somehow (either explicitly or a default param)